src/components/TooltipNew/GuideTooltip.js

```diff
@@ -0,0 +1,272 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './style.scss';
+import PopperTooltip from './PopperTooltip';
+import TooltipPortal from './TooltipPortal';
+
+const baseClass = 'guide-tooltip';
+
+const cx = classNames.bind(styles);
+
+const addPadding = (rect, padding) => {
+  const x = Math.round(rect.x) - padding;
+  const y = Math.round(rect.y) - padding;
+  const width = Math.round(rect.width) + 2 * padding;
+  const height = Math.round(rect.height) + 2 * padding;
+  const top = y;
+  const left = x;
+  const bottom = top + height;
+  const right = left + width;
+
+  return {
+    x,
+    y,
+    width,
+    height,
+    top,
+    left,
+    bottom,
+    right
+  };
+};
+
+class VirtualReference {
```

Comment:
move it to global helpers (directory next to components) - it will be useful in other components. `addPadding` could be a method of this class. Add `padding: number` argument to the constructor.

Reply:
ok